### DISCUSSION TITLE
how to diff one device from anothor device

### DISCUSSION THREAD

**Discussion by wangzy0327** [intel/llvm#7556] [Q&A]
**Describe the bug**
If I have two same spec devices. How to differ them ? It seems no device_id or platform_id to differ two device(same specification)

![42043f91e90323be7bc2bc5fa51b661](https://user-images.githubusercontent.com/22990858/204271355-eda1cd33-55ac-4cf7-952b-c196ac7eb0fd.png)

**Comment by JackAKirk** [comment]
If only Nvidia gpus are available, a list of all available devices can be found from:
```
std::vector<sycl::device> Devs =
      sycl::device::get_devices(sycl::info::device_type::gpu);
```

The ordering of the list will match the ordering of the devices reported by the nvidia driver using:
```
$ nvidia-smi
```
i.e device 0 reported by `nvidia-smi` will map to Devs[0] above (if only Nvidia GPUs are visible)

If GPUs from other backends are available you need some more information:

Each CUDA device currently has its own platform. The vector of platforms returned from sycl::platform::get_platforms() is ordered by CUDA device ID with the lowest ID available coming first. This means that we can populate an identical list of devices, `devs`, as above in the following way:
```
std::vector<sycl::device> Devs;

for (const auto &plt : sycl::platform::get_platforms()) {

  if (plt.get_backend() == sycl::backend::cuda)
    Devs.push_back(plt.get_devices()[0]);
}
```

NOTE: In the long-term CUDA devices will be contained within a single platform, and the above code will have to be adjusted.

**Comment by zjin-lcf** [comment]
I think users can query device id or uuid.
